request.go

```diff
@@ -291,6 +292,37 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
 					models = reflect.Append(models, m)
 				}
 
+				if links != nil {
+					var assignedValue bool
+					linkModel := reflect.New(fieldValue.Type().Elem().Elem())
+					linkModelValue := linkModel.Elem()
+					linkModelType := linkModelValue.Type()
+
+					for i := 0; i < linkModelValue.NumField(); i++ {
+						fieldType := linkModelType.Field(i)
+						tag := fieldType.Tag.Get("jsonapi")
+						if tag == "" {
+							continue
+						}
+						fieldValue := linkModelValue.Field(i)
+						args := strings.Split(tag, ",")
+						annotation := args[0]
+						if annotation == annotationLinks {
+							value, err := unmarshalAttribute(*links, args, fieldType, fieldValue)
+							if err != nil {
+								er = err
+								break
+							}
+
+							assign(fieldValue, value)
+							assignedValue = true
+						}
+					}
+					if assignedValue {
+						models = reflect.Append(models, linkModel)
```

Comment:
We are appending here because we are operating when the current relationship is a list of relations.